### ISSUE TITLE
`spirv-val` Does Not Catch 64-bit Related Errors While Validation Layer Does

### ISSUE THREAD

**Issue by cwfitzgerald** [KhronosGroup/Vulkan-ValidationLayers#7542]
**Environment:**
 - OS: Mac, but appears elsewhere
 - SDK or header version if building from repo: 1.3.268
 - Options enabled (synchronization, best practices, etc.): None

**Describe the Issue**

The following spirv is invalid per `VUID-StandaloneSpirv-Base-04781`;

```spirv
%long = OpTypeInt 64 1
%18 = OpLoad %long %17
%19 = OpBitReverse %long %18
```

When I run such a spir-v against the validation layers, it throws the following error.

```
[2024-02-20T17:32:43Z ERROR wgpu_test::expectations] Validation Error: Validation Error: [ VUID-VkShaderModuleCreateInfo-pCode-08737 ] | MessageID = 0xa5625282 | vkCreateShaderModule(): pCreateInfo->pCode is not valid SPIR-V: [VUID-StandaloneSpirv-Base-04781] Expected 32-bit int type for Base operand: BitReverse
      %19 = OpBitReverse %long %18
    . The Vulkan spec states: If pCode is a pointer to SPIR-V code, pCode must adhere to the validation rules described by the Validation Rules within a Module section of the SPIR-V Environment appendix (https://vulkan.lunarg.com/doc/view/1.3.268.1/mac/1.3-extensions/vkspec.html#VUID-VkShaderModuleCreateInfo-pCode-08737)
```

However when I run [the full shader](https://github.com/KhronosGroup/Vulkan-ValidationLayers/files/14350289/test.spvasm.txt) against `spirv-val` with any target-env, it passes with flying colors.

Is there a way to get these validation issues from `spirv-val` without needing to actually try to build a pipeline?

**Comment by cwfitzgerald** [comment]
Oh interesting, how does the validation layers go and find it? It's not impossible that the "spirv-val" i have in my path is different from the one the validation layers are finding

**Comment by spencer-lunarg** [comment]
We build SPIR-V and link into the dynamic library version of `spirv-val`, it doesn't use the executable 

the exact version is found in https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/scripts/known_good.json

I guess I would recommend the latest 1.3.275 SDK as the version I tested with in the above comment

**Comment by spencer-lunarg** [comment]
@cwfitzgerald happy to keep helping, but since there is nothing to do inside the Validation Layers, and things are working as expected from what I see, I am going to close this issue
